InfinityCosmos/ForMathlib/AlgebraicTopology/SimplicialSet/MorphismProperty.lean

```diff
@@ -26,4 +27,35 @@ def trivialFibration : MorphismProperty SSet := fun _ _ p ↦ BoundaryInclusions
 
 end trivialFibration
 
+section isoFibration
+
+/-- Inductive definition of inner horn inclusions Λ[n, i] ⟶ Δ[n] 
+  by restricting general horn inclusions to 0 < i < n -/
+inductive InnerInclusion : {X Y : SSet} → (X ⟶ Y) → Prop
+  | mk (n i : ℕ) (low : 0 < i) (high : i < n) : InnerInclusion (hornInclusion n i)
+
+/-- The class of inner horn inclusions as a MorphismProperty -/
+def InnerInclusions : MorphismProperty SSet := fun _ _ p ↦ InnerInclusion p
+
+/-- Inductive definition of being equal to the coherent iso -/
+inductive CoherentIsoInclusion : {X Y : SSet} → (X ⟶ Y) → Prop
+  | mk : CoherentIsoInclusion (coherentIso.pt .zero)
+
+/-- The class of the inclusion into the coherent iso as a MorphismProperty -/
+def IsCoherentIso : MorphismProperty SSet := fun _ _ p ↦ CoherentIsoInclusion p
+
+/-- The union of inner horn inclusions and inclusion into coherent iso -/
+def InnerIsoInclusion {X Y : SSet} (p : X ⟶ Y) : Prop :=
+  (InnerInclusions p) ∨ IsCoherentIso p
+
+/-- The union of the class of inner horn inclusions and the inclusion into coherent iso 
+  as a MorphismProperty -/
+def InnerIsoInclusions : MorphismProperty SSet := fun _ _ p ↦ InnerIsoInclusion p
+
+/-- Definition of isofibration: An isofibration of quasi-categories is a morphism that
+has the rlp wrt inner horn inclusions and the inclusion into the walking coherent iso. -/
+def isoFibration : MorphismProperty SSet := fun _ _ p ↦ InnerIsoInclusions.rlp p
```

Comment:
I only like this definition when the ambient simplicial sets are quasi-categories (suffices, and I think is better, to assume this just for the codomain). Can that be added to the morphism property api somehow? (If not, ask @mckoen and others about this on zulip).

Reply:
I think if there is a category of quasi-categories, then it should be possible to use MorphismProperty directly, however, if you want just one of them, then yeah I probably would have to ask.

Reply:
FYI @emilyriehl the definition you mentioned here, does not match with the latex file, where it says "between quasi-categories". Should that be adjusted accordingly?

Reply:
I like the solution of stating this definition for the category of quasi-categories only. Then we can apply the morphism property api directly without modification. Would you mind adding that subcategory for us, perhaps called QCat? It will be useful anyway. (NB: I'm pretty sure Lean knows about full subcategories.)